Automattic/jetpack
# Add @automattic/jetpack-user package

related: https://github.com/Automattic/jetpack/pull/20543

Moves minimal user-related JS from the Jetpack Admin app into the `@automattic/jetpack-user`.

This was extracted out of the [Standalone My Plans Page PR](https://github.com/Automattic/jetpack/pull/20543).

In time this 

#### Changes proposed in this Pull Request:
* Add `@automattic/jetpack-user` package

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

Nothing should change. `jetpack build plugins/jetpack` and confirm that the UI operates without errors.
